File: src/data/courses/dsa/linked-lists/reversal-segment/reverse-first-k/code/java/ReverseFirstK.java
// ============================================================
// Reverse First K Nodes
// ============================================================
class Solution {
    public ListNode reverseFirstK(ListNode head, int k) {
        if (head == null || k <= 1) return head;
        ListNode prev = null, curr = head, orig = head;
        for (int i = 0; i < k && curr != null; i++) {
            ListNode next = curr.next; curr.next = prev; prev = curr; curr = next;
        }
        orig.next = curr; return prev;
    }
}

File: src/data/courses/dsa/linked-lists/reversal-segment/reverse-first-k/code/python/reverse_first_k.py
# ============================================================
# Reverse First K Nodes
# ============================================================
class Solution:
    def reverseFirstK(self, head, k):
        if not head or k <= 1: return head
        prev, curr, orig = None, head, head
        for _ in range(k):
            if not curr: break
            nxt = curr.next; curr.next = prev; prev = curr; curr = nxt
        orig.next = curr  # Connect to remaining
        return prev
